test/gtest/shp/copy.cpp

```diff
@@ -4,42 +4,144 @@
 
 #include "shp-tests.hpp"
 
-using T = int;
-using DV = shp::distributed_vector<T, shp::device_allocator<T>>;
-using V = std::vector<T>;
+template <typename AllocT> class CopyTest : public testing::Test {
+public:
+  using DistVec = shp::distributed_vector<typename AllocT::value_type, AllocT>;
+  using LocalVec = std::vector<typename AllocT::value_type>;
+};
 
-TEST(ShpTests, Copy_Dist2Local) {
-  const int n = 100;
-  std::size_t n_to_copy = 20;
+using AllocatorTypes =
+    ::testing::Types<shp::device_allocator<int>,
+                     shp::shared_allocator<long long unsigned int>>;
+TYPED_TEST_SUITE(CopyTest, AllocatorTypes);
```

Comment:
This is nice. Can we do share tests for `shp::distributed_vector` and `mhp::distributed_vector` this way?

Reply:
Yes, we can.
But let's do it in separate PR as it is not yet obvious to me how to do it in best way.